pkgs/swiftgen/swift2objc/lib/src/parser/_core/utils/parse_symbol_id.dart

```diff
@@ -0,0 +1,5 @@
+import '../type_defs.dart';
+
+String parseSymbolId(JsonMap symbolJson) {
+  return symbolJson["identifier"]["precise"];
```

Comment:
What happens if the json is malformed and one of these keys doesn't exist or is the wrong type? It'll throw some sort of error, but are you catching that error somewhere? That'll be pretty confusing to users.

Maybe catch these errors at the top level of the parse and log them? Or wrap them in a custom exception class (eg `SymbolgraphError` or something) and rethrow them?

Reply:
I was focusing on getting things to work so I assumed the json is correct (since it's generated by a tool anyway). How much info do u suggest to capture in the error? Do we just log "invalid-json" or specify exactly what's wrong with it?

Reply:
There's 2 kinds of people that will see this exception, users, and us.

For the users, we want to tell them "this json file contains stuff we don't recognise" (ie, a more specific error than NPE or whatever, which just makes it look like our code is buggy).

For us, we need there to be enough information logged that we can debug it when users file a bug, or when we see it during development. So a stack trace would be good, as well as whatever info is in the original error.

So I'd catch the exception this throws, wrap it in a custom exception that indicates its some sort of problem with the json, then `rethrow` it. That way we get all the info of the original error and stack trace, and users see our custom exception that says there's something wrong with their json. Make sense?

Reply:
It looks like there's a bunch of these sorts of unchecked lookups throughout the parsers. I wonder if there's a way of factoring out the error check?

You could add extension methods to `JsonMap` called `getMap` that returns a `JsonMap` and `getString` returns a string, These methods would the getting and the type cast, and throw the custom error if something goes wrong. This approach would also be more strongly typed and clear.

Reply:
I guess you'll also need a `getList` function that returns a `List<JsonMap>`?

Reply:
Well since we added a new Json class to handle all of this I guess we can mark this as resolved